File: array/sort.cpp
#include <iostream>

using namespace std;

// implement bubble sort
void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
            }
        }
    }
}

// implement insertion sort
void insertionSort(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        int key = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }
}

// implement maximum sort
void maximumSort(int arr[], int n) {
    int key = n-1;
    for (int i = 0; i < n - 1; i++) {
        int maks = arr[0];
        int imaks = 0;
        for (int j = 1; j <= key; j++) {
            if (arr[j] > maks) {
                maks = arr[j];
                imaks = j;
            }
        }
        swap(arr[key], arr[imaks]);
        key--;
    }
}

int main() {
    int arr[] = {9, 3, 2, 1, 5, 6, 7, 8, 10};
    int arr2[] = {9, 3, 2, 1, 5, 6, 7, 8, 10};
    int arr3[] = {9, 3, 2, 1, 5, 6, 7, 8, 10};

    int n = sizeof(arr) / sizeof(arr[0]);

    // bubble sort
    cout << "Bubble sort: "; bubbleSort(arr, n);
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << '\n';

    // insertion sort
    cout << "Insertion sort: "; insertionSort(arr2, n);
    for (int i = 0; i < n; i++) {
        cout << arr2[i] << " ";
    }
    cout << '\n';

    // maximum sort
    cout << "Maximum sort: "; maximumSort(arr3, n);
    for (int i = 0; i < n; i++) {
        cout << arr3[i] << " ";
    }
    cout << '\n';

    return 0;
}

File: array/sort.py
# implement bubble sort
def bubble_sort(arr):
    for i in range(len(arr)):
        for j in range(len(arr)-1-i):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr

# implement insertion sort
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i-1
        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
    return arr

# implement maximum sort
def max_sort(arr):
    key = len(arr) - 1
    for i in range (0, len(arr)-1):
        max = arr[0]
        imax = 0
        for j in range(1, key+1):
            if (arr[j] > max):
                max = arr[j]
                imax = j
        arr[key], arr[imax] = arr[imax], arr[key]
        key -= 1
    return arr

arr = [9, 3, 2, 1, 4, 3, 10]

# print(bubble_sort(arr))
# print(insertion_sort(arr))
# print(max_sort(arr))

print(arr)



